--- bb-1.3.0/timers.c.pulse	2017-06-30 21:37:27.711687696 +0200
+++ bb-1.3.0/timers.c	2017-06-30 21:46:36.218568976 +0200
@@ -483,14 +483,17 @@ tl_process_group (tl_group * group, int
 	      int time = timer->interval - tl_lookup_timer (timer);
 	      if (time <= 0)
 		{
+		  int n;
 		  if (activated != NULL)
 		    (*activated)++;
 		  again = 1;
 		  tl_reset_timer (timer);
 		  if (time < -200 * 1000000)
 		    time = 0;	/*underflow? */
-		  tl_slowdown_timer (timer, time);
-		  time = timer->interval + time;
+		  n = -(time) / timer->interval + 1;
+		  time += timer->interval * n;
+
+		  tl_slowdown_timer (timer, time - timer->interval);
 		  timer->handler (timer->userdata);
 		  tl_update_time ();
 		}
